libshaderc_spvc/include/spvc/spvc.hpp

```diff
@@ -361,6 +362,45 @@ class Context {
     return shaderc_spvc_set_decoration(context_.get(), id, decoration, argument);
   }
 
+  // Get spirv_cross decoration (added for GLSL API support in Dawn).
+  // Given an id and a decoration, result is sent out through |argument|
+  // if |id| does not exist, returns an error message.
+  shaderc_spvc_status GetDecoration(uint32_t id,
+                                    shaderc_spvc_decoration decoration,
+                                    uint32_t* argument){
+    return shaderc_spvc_get_decoration(context_.get(), id, decoration, argument);
+  }
+
+  // Unset spirv_cross decoration (added for GLSL API support in Dawn).
+  // Given an id and a decoration. Assuming id is valid.
+  shaderc_spvc_status UnSetDecoration(uint32_t id,
+      shaderc_spvc_decoration decoration){
+    return shaderc_spvc_unset_decoration(context_.get(), id, decoration);
+  }
+
+// For each combined_image_sampler calls the provided callback function |f|,
```

Comment:
nit: For each combined_image_sampler, instead of For each combined_image_sampler